### PR TITLE
Upgraded Bootstrap 2 to 2.3.2

### PR DESCRIPTION
**Pull request by biyeun**
And (as a requirement), upgraded jquery to latest for hq and jquery ui to 1.8.23
placed on staging to see what it churns up

@dannyroberts @benrudolph @TylerSheffels 
cc @millerdev @czue

### CODE REVIEW

File: corehq/apps/style/management/commands/fix_less_imports_collectstatic.py [hunk 1]
```diff
@@ -21,24 +21,29 @@ def handle(self, *args, **options):
...
-                        new_path = settings.LESS_B3_PATHS.get(path_file)
-                        if new_path is not None:
-                            print ("[CORRECTING] %(f_name)s path in %(less_file)s: "
-                                   "%(new_path)s" % {
-                                       'f_name': path_file,
-                                       'less_file': less_file,
-                                       'new_path': new_path,
-                                   })
-                            content = re.sub(B3_REGEX, new_path, content)
+                    BOWER_PATH =  '../../../../../../bower_components'
```
Comment (dimagimon) [inline corehq/apps/style/management/commands/fix_less_imports_collectstatic.py]: E222 multiple spaces after operator

**Comment by biyeun** [comment]
+206 −21,838 that's right. :cake:

File: corehq/apps/style/management/commands/fix_less_imports_collectstatic.py (hunk 1 above)
Comment (biyeun) [inline corehq/apps/style/management/commands/fix_less_imports_collectstatic.py]: yo fu dimagimon

**Comment by czue** [comment]
:birthday:

File: corehq/apps/style/management/commands/fix_less_imports_collectstatic.py (hunk 1 above)
Comment (benrudolph) [inline corehq/apps/style/management/commands/fix_less_imports_collectstatic.py]: haha. can we use an absolute path here? `os.path.join(settings.BASE_DIR, bower_components)`

Comment (benrudolph) [inline corehq/apps/style/management/commands/fix_less_imports_collectstatic.py]: nvm realize why we need this

**Comment by benrudolph** [comment]
so many awesome things in this PR! woohoo

**Comment by millerdev** [comment]
woot!

**Comment by TylerSheffels** [comment]
This is awesome, and great clean up of things that confused me a lot!

File: corehq/apps/style/management/commands/fix_less_imports_collectstatic.py (hunk 1 above)
Comment (biyeun) [inline corehq/apps/style/management/commands/fix_less_imports_collectstatic.py]: yeah absolute path won't work

**Comment by dannyroberts** [comment]
Awesome, if you merge in `jquery-ui-1.8.23-legacy` you'll also get a resolution for jquery in there

**Comment by biyeun** [comment]
@czue @benrudolph @millerdev the plan is to leave this on staging until wednesday to filter out any lingering issues. Thoughts?

**Comment by biyeun** [comment]
@dannyroberts that upgrade is already a part of this pr...was necessary to get jquery 1.9 running which was necessary to get 2.3.2 running

**Comment by benrudolph** [comment]
cool i'm game. we're kind of stuck between a rock and a hard place until we get proper error reporting and testing setup

**Comment by czue** [comment]
it's really hard for me to know how good we expect "it being on staging" to be in terms of finding potential issues, or what the possible scope of bugs that this PR could introduce is.

to a point @calellowitz made recently - staging isn't used enough to expect it to be a good representation of everything that happens in HQ. so I think we should expect that it being on staging means that maybe 15% of the functionality on HQ will get exercised (but probably the most important 15%).

if you and the other JS gurus are comfortable with that number given the possible failure modes then wednesday sounds fine. if you aren't then maybe doing an explicit QA pass or deploying to india first (maybe over the weekend) would be safer. i don't feel like i know enough of about how this could go wrong to have a useful opinion.

**Comment by benrudolph** [comment]
good thoughts. i think we should approach this very similarly to a python package update that we use site wide. the things we do for that usually consist of reading change logs, putting on staging, ensuring tests pass, and clicking around. some of those things don't apply to our frontend stack because it's still maturing. the only thing that worries me about the QA request to do the whole site is the cost/benefit ratio. i remember @christineawhite saying it would take valley labs a pretty long time to do an entire site QA. maybe there should be UI test plan that's less comprehensive, but ensures that important buttons and UI elements are working

i feel like since we're using the jquery migrate package, that should mitigate any serious jquery incompatibilities. as for the bootstrap upgrade, i think @biyeun would have the best insight on the changes from the upgrade and the risks they pose

**Comment by christineawhite** [comment]
@benrudolph keep me posted, we may have some free time over the holidays between xmas and new years that we can have VLs do a full scrub of HQ

**Comment by millerdev** [comment]
I agree with @czue. Unless you're explicitly testing things yourself, I would not expect them to get tested on staging. A lot of times when I hit a bug on staging I'm likely to try to discover what I want another way rather than digging into the bug itself. I'll dig into the bug or report an issue as a last resort. I assume other people do the same thing since staging is kinda the wild west/disaster zone, and it's distracting to dig into a bug that seems unrelated to one's current task.

I think a limited QA run would be worth doing in this case.

**Comment by biyeun** [comment]
@czue @millerdev @benrudolph so I did a fairly thorough test of things that I thought would break from this, namely

- reporting
- app builder
- complex forms in Reminders (although that's moving to B3 soon anyway, but was good to test)
- case list / case history
- submit history / form information

These I think were the areas that could have caused bugs. I registered no javascript errors and only a few minor css issues with the help text (which I then fixed). I will be hyper vigilant over the next few days once this is merged to look out for issues and respond to them, but my feeling is that the biggest issues will be tweaks in CSS, as the heavy JS pieces seemed to check out fine on staging.

If there are any issues, it's likely that they'll come from edge cases rather than core workflows. If you guys are cool with this then I'll lift the label and get this merged

**Comment by dannyroberts** [comment]
I'm down with this plan. (The fact that you didn't find anything horribly wrong during your manual QA + the fact that this is a minor release which *should* mean backwards compatible makes me feel pretty good about this plan.)

**Comment by biyeun** [comment]
cool, then let's get this out and knock off one of our security to-dos!

**Comment by millerdev** [comment]
awesome!

**Comment by TylerSheffels** [comment]
Nicely done!

**Comment by czue** [comment]
sounds good to me! :birthday: :cake: :fish_cake:

**Comment by benrudolph** [comment]
:boom:

**Pull request merged**